posts/mixtral/index.ipynb

```diff
@@ -0,0 +1,259 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "1669e852-3e4c-4819-a9ee-bfedd2ab62ed",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "title: \"Build a Mixtral Chatbot with Panel\"\n",
+    "date: \"2023-12-12\"\n",
+    "description: \"With Transformers and llama.cpp\"\n",
+    "author:\n",
+    "  - Andrew Huang\n",
+    "  - Philipp Rudiger\n",
+    "  - Sophia Yang\n",
+    "categories: [showcase, panel]\n",
+    "image: \"images/chatbot.png\"\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eb68fb53-5c93-449a-bf16-edced92ff738",
+   "metadata": {},
+   "source": [
+    "Mistral AI just announced the Mixtral 8x7B and the Mixtral 8x7B Instruct models. These models have shown really amazing performance, outperforming Llama 2 and GPT 3.5 in many benchmarks. They've quickly became the most popular 7B parameter open weights models in the AI world. In this blog post, we will walk you through how to build AI chatbots with the Mixtral 8x7B Instruct model using the Panel chat interface. We will cover two methods: \n",
+    "\n",
+    "- Method 1: Run Mixtral with transformers (A100 required)\n",
+    "- Method 2: Run Mixtral with llama.cpp (can run on Macbook)\n",
+    "\n",
+    "# Method 1: Run Mixtral with transformers (A100 required)\n",
+    "\n",
+    "The first method is to use the latest Transformers from HuggingFace. We adapted the code from this blog post: [https://huggingface.co/blog/mixtral](https://huggingface.co/blog/mixtral). \n",
+    "\n",
+    "The following code: \n",
+    "\n",
+    "- Defines the model `mistralai/Mixtral-8x7B-Instruct-v0.1`\n",
+    "- Uses the Tokenizer from this model to format the input user message\n",
+    "- Uses the transformers pipeline to specify the text-generation pipeline, the model, and the 4-bit quantization.\n",
+    "- Uses the `TextStreamer` to stream the text output \n",
+    "\n",
+    "```python\n",
+    "# REF: Code adapted from https://huggingface.co/blog/mixtral\n",
+    "!pip install -U \"transformers==4.36.0\"  --upgrade\n",
+    "!pip install accelerate\n",
+    "!pip install bitsandbytes\n",
```

Comment:
Maybe replace these commands by some text saying these packages need to be installed? Not sure it's good advice to have users run those in their environment (you're running it in Colab maybe?).

Reply:
hmmm so I took the code from the HF blog and they used !pip xxx...